src/App.tsx

```diff
@@ -1,10 +1,122 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loader, setLoader] = useState<number | null>();
+
+  const handleErrorMessage = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(response => {
+        let filteredTodos = response;
+
+        if (filter === 'active') {
+          filteredTodos = response.filter(todo => todo.completed === false);
+        }
+
+        if (filter === 'completed') {
+          filteredTodos = response.filter(todo => todo.completed === true);
+        }
+
+        setTodos(filteredTodos);
+        setErrorMessage('');
+      })
+      .catch(() => {
+        handleErrorMessage('Unable to load todos');
+      });
+  }, [filter]);
```

Comment:
```suggestion
  useEffect(() => {
    getTodos()
      .then(response => {
        setTodos(response);
        setErrorMessage('');
      })
      .catch(() => {
        handleErrorMessage('Unable to load todos');
      });
  }, []);
  
  const filteredTodos = getFilteredTodos(todos, filter)
```

to avoid redundant fetch requests